### PR TITLE
[RPC] Use qualified name str directly in RPC torch script code path

### PR DESCRIPTION
**Pull request by xush6528**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34733 [RPC] Use qualified name str directly in RPC torch script code path**

simplify

Differential Revision: [D20442573](https://our.internmc.facebook.com/intern/diff/D20442573/)

### CODE REVIEW

File: torch/csrc/distributed/rpc/init.cpp [hunk 1]
```diff
@@ -421,17 +421,16 @@ If the future completes with an error, an exception is thrown.
   module.def(
       "_invoke_rpc_torchscript",
       [](const std::string& dstWorkerName,
-         const py::object& userCallable,
+         const std::string& qualifiedNameStr,
          const py::tuple& argsTuple,
          const py::dict& kwargsDict) {
```
Comment (zhaojuanmao) [inline torch/csrc/distributed/rpc/init.cpp]: nit: change them back to py::args and py::kwargs?

**Review by mrshenli — COMMENTED** [review]
Agree with @zhaojuanmao  on arg type consistency. Others LGTM!

Could you please rebase, test failures should have been fixed.

**Review by mrshenli — APPROVED** [review]
Thanks for adding this!

File: torch/csrc/distributed/rpc/init.cpp (hunk 1 above)
Comment (xush6528) [reply torch/csrc/distributed/rpc/init.cpp]: Changing this will cause memory leak. I will re-land the rest and debug on this later.

**Pull request closed without merging**